docs/source/extensions_tutorial/7_publishing_extensions.rst

```diff
@@ -1,4 +1,114 @@
+
+
+Publish your extension on PyPI
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+To make your extension installable for users via pip and manage public releases NWB uses the
+`Python Package Index (PyPI) <https://pypi.org/>`_ index.
+
+1. Follow `these directions <https://packaging.python.org/tutorials/packaging-projects/>`_ to package your project.
+   You may need to modify ``setup.py``. If your extension version is ``0.1.0``, then this page
+   should exist: ``https://pypi.org/project/<myextension>/0.1.0``
+2. Once your GitHub release and ``setup.py`` are ready, publishing on PyPI:
+
+.. code-block:: bash
+
+    python setup.py sdist bdist_wheel
+    twine upload dist/*
+
+Publishing your extension on the NDX Catalog
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The :ndx-catalog:`NDX Catalog <>` serves as a central, community-led catalog for extensions to
+the NWB data standard. The NDX Catalog manages basic metadata about extensions while ownership of the source
+repositories for the extensions remain with the developers. To publish your extension on the catalog:
+
+1. Fork the `staged-extensions <https://github.com/nwb-extensions/staged-extensions> repository.
+   The ``staged-extensions`` repository is used to submit new extension to the catalog via pull requests.
+2. Clone your fork of the ``staged-extensions`` onto your local computer
+3. Copy the directory ``staged-extensions/example`` to a new directory with the name of your extension:
+
+.. code-block:: bash
+
+    cp -r staged-extensions/example staged-extensions/<my_extension>
+
+4. Edit ``staged-extensions/<my_extension>/ndx-meta.yaml`` with information on where to find your
+   NWB extension. The ``NEXTSTEPS.md`` file in the ``ndx-template`` includes an autogenerated
+   template ``ndx-meta.yaml`` file that you may copy and modify. The YAML file MUST contain a dict
+   with the following keys:
+
+    * ``name``: extension namespace name
+    * ``version``: extension version
+    * ``src``: URL for the main page of the public repository (e.g. on GitHub, BitBucket, GitLab)
+      that contains the sources of the extension
+    * ``pip``: URL for the main page of the extension on PyPI
+    * ``license``: name of the license of the extension
+    * `` maintainers``: list of GitHub usernames of those who will reliably maintain the extension
+       You may copy and modify the following YAML that was auto-generated:
+
+5. Edit ``staged-extensions/<my_extension>/README.md`` to add information about your extension.
+   Usually, you can here just copy the ``README.md`` from your extension repo
+   ``cp <my_extension>/README.md staged-extensions/<my_extension>/README.md``
+
+6. Add and commit your changes to Git and push your changes to GitHub:
+
+.. code-block::
+
+    cd staged-extensions
+    git add <my_extension>
+    git commit -m "Add new catalog entry for <my_extension>"
+    git push
+
+7. Open a pull request. Building of your extension will be tested on Windows, Mac, and Linux.
```

Comment:
This step might be confusing to users who haven't had experience with github. Is there a link we can provide?

Reply:
I added a corresponding link to GitHub docs. I also added a screenshot to show how to clone and fork repos.